### DISCUSSION TITLE
Pitch change and Mixing of audio file.

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#31] [General]
@marcel-licence  I am planning to make an electronic version of a drone instrument called 'TANPURA' which is used in Indian classical music. It basically has 4-strings and each one has a different pitch, and an overall pitch(note) which should be changeable. Also the the pitch of the first string must be changeable regardless of the pitch of other strings and overall pitch. The strings are played one by one, but since the release parameter of the string is higher a string tone will overlap to the next one which would require the mixing of sound.
I am thinking of storing the tone of the instrument in wav or hex format and using an ESP32  to do the job for me. Please help me to know how did you achieve the changing of  the pitch of the stored audio file and mixing of sound(Polyphony)using ESP32.

**Comment by marcel-licence** [comment]
You should look for the functions also referenced in z_config.ino.
For example following will help:

Sampler_LoopStartC
Sampler_LoopStartF
Sampler_LoopEndC
Sampler_LoopEndF

Sampler_SetLoopEndMultiplier
Sampler_SetPitch

**Reply by souravlal** [reply]
Let me try.

**Reply by souravlal** [reply]
where are these functions do their job in the code.
I have to load one sample and change its pitch and sometimes mix the different notes of that sample when certain buttons are pressed and play it through the internal DAC of ESP32 can anyone help me to accomplish this task?
